backend/logger/fileHandler.py

```diff
@@ -0,0 +1,80 @@
+import os
+import uuid
+import datetime
+import atexit
+
+
+class FileHandler:
+    """
+    In charge of handling files for the logger. This class will open, close,
+    and name files for the logger. Responsible for storing logs in safe place.
+    In addition, it will pass file handles to different programs.
+    """
+
+    def __init__(self, file_name):
+        self.file_name = file_name
+        self.full_name = file_name
+        self.file_handle = None
+
+        atexit.register(self.closeFile)
+
+    def createName(self):
+        """Create and return the name of the file.
+        The format is filename-uid-date
+
+        Returns:
+        name (str) : the name of the file
+        """
+        date = datetime.date.today().strftime("%b-%d-%Y")
+        uid = str(uuid.uuid4())
+        self.full_name = self.file_name + "-" + uid + "-" + date + ".log"
+        return self.full_name
+
+    def getFileLocation(self):
+        """Return the location in which the file should be
+        stored. Print this to stdout as well. Return value
+        is platform-dependent
+
+        Returns:
+        location (str) : location log will be stored
+        """
+
+        curr_dir = os.getcwd()
+        if os.name == "nt":
```

Comment:
I assume `nt` means windows, but could you add a comment to specify which OS it is?

Reply:
Yes nt is the family of OS's produced by Microsoft since 1993. Will add the comment.

Reply:
I think you should use something like os.path.join to avoid this sort of handling for different operating systems.

Reply:
You should also just be able to use a relative path and not need to prepend curr_dir

Reply:
I wasn't sure about how we would keep the structure of our files. This is something I'm still not 100% sure about. I was thinking about writing all the logs to $MAIN_DIR/backend/logs regardless of where /Logger may move in the future. The current implementation has to change either way, but I don't know if relative path is the answer.